### PR TITLE
sinsp_observer

### PR DESCRIPTION
**Pull request by deepskyblue86**
**What type of PR is this?**

> /kind cleanup

**Any specific area of the project related to this PR?**

> /area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:
Introduce `sinsp_observer`, taking over `sinsp_fd_listener`, and have it at `sinsp` level, plus some minor cleanups.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
refactor(userspace/libsinsp)!: introduce sinsp_observer, taking over sinsp_fd_listener
update(userspace/libsinsp): track_connection_status getter/setter in sinsp
update(userspace/libsinsp): introduce sinsp_evt::get_errorcode()
update(userspace/libsinsp): make sinsp_fdinfo all public and drop friends
refactor(userspace/libsinsp)!: move sinsp_parser::set_unix_info to sinsp_fdinfo
```

### CODE REVIEW

File: userspace/libsinsp/parsers.cpp [hunk 1]
```diff
@@ -4015,11 +4007,7 @@ bool sinsp_parser::update_fd(sinsp_evt *evt, sinsp_evt_param *parinfo)
 	else if(family == PPM_AF_UNIX)
 	{
 		evt->m_fdinfo->m_type = SCAP_FD_UNIX_SOCK;
-		if(set_unix_info(evt->m_fdinfo, packed_data) == false)
-		{
-			return false;
-		}
-
+		evt->m_fdinfo->set_unix_info(packed_data);
```
Comment (FedeDP) [inline userspace/libsinsp/parsers.cpp:4010]: Can you explain this change?

Comment (FedeDP) [reply userspace/libsinsp/parsers.cpp:4010]: Oh i see, it was always returning true so you made it `void`. Thanks!

**Review by FedeDP — APPROVED** [review]
I love this cleanup!
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**